models/Mapper/ItemVariableMapper.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoResultServer\models\Mapper;
+
+use oat\oatbox\service\ConfigurableService;
+
+class ItemVariableMapper extends ConfigurableService
+{
+    public function splitByAttempt(array $itemVariables): array
```

Comment:
```suggestion
namespace oat\taoResultServer\models\format;

use oat\oatbox\service\ConfigurableService;

class ItemResponseVariableSplitter extends ConfigurableService
{
    public function split(array $itemVariables): array
```

This is actually not mapping an object, but a primitive type, so I would not call it mapper. Also it is splitting an array into multiple, but not changing it. You can make the name generic `split`, cause if we want to split response based on other criteria besides attempts this logic keeps inside this class.

What do you think?